### PR TITLE
Support CoffeeScript v2

### PR DESCRIPTION
**Pull request by etamponi**
This PR is an attempt to support CoffeeScript v2.

All tests pass, and I've tested it on one of my services just ported to CoffeeScript v2, and it runs fine.

I must admit the changes will seem... hacky. And they totally are :P but at least it runs, so perhaps people more experienced than me with compilers and ASTs might start from here to make this a real thing.

### CODE REVIEW

File: test/istanbul/pragmaTests.coffee [hunk 1]
```diff
@@ -85,7 +85,7 @@ module.exports = (run) ->
             istanbulStyle = """
                 console.log "hello"
                 ### istanbul ignore else ###
-                if x
+                if x is true
```
Comment (jwalton) [inline test/istanbul/pragmaTests.coffee]: Why this change?

Comment (etamponi) [reply test/istanbul/pragmaTests.coffee]: It's for test coverage: the comments get assigned to a different token depending on how many elements follow the `if` keyword

**Comment by jwalton** [comment]
This all looks fine.  Do you know why this is failing in travis right now though?  https://travis-ci.org/benbria/coffee-coverage/builds/303496827?utm_source=github_status&utm_medium=notification

**Comment by etamponi** [comment]
It looks like Travis cannot find `coffee`, it is failing with a ENOENT and a `sh: 1: coffee: not found` error

**Comment by etamponi** [comment]
Perhaps I know why, I'll try to push a small change

**Comment by jwalton** [comment]
Looks like it's the "preinstall" script.  If I clone your repo and `npm install`, same thing happens.

**Comment by jwalton** [comment]
Oh yeah, you found it.  :)

**Comment by jwalton** [comment]
Now it's failing on node 4 and node 5, but I think it's coffee-script that's failing.  Ok, I'll merge this.

**Comment by jwalton** [comment]
Published as v3.0.0.  Thanks!

File: .travis.yml [hunk 2]
```diff
@@ -7,7 +7,7 @@ node_js:
 # Install latest npm, because node v0.8 comes with npm v1.2.30, which doesn't like `^`s in
 # package.json.
 before_install:
-  - "npm install -g npm@'>=1.4.3'"
+  - "npm install -g npm@'>=1.4.3' coffee-script"
```
Comment (GeoffreyBooth) [inline .travis.yml]: CoffeeScript on NPM has moved to `coffeescript`.

**Comment by GeoffreyBooth** [comment]
> Now it’s failing on node 4 and node 5

CoffeeScript 2 is Node 6+.

**Comment by etamponi** [comment]
I got rid of that dependency in the following commit, anyway it's not used
in coffee-coverage 3.0 because it uses another commit

Il 14 dic 2017 00:40, "Geoffrey Booth" <notifications@github.com> ha
scritto:
